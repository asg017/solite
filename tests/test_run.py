def test_run_basic(solite_cli, snapshot, tmp_path):
    (tmp_path / "a.sql").write_text(
        """
.timer off

select 1;

select 2;

select 3
""",
        newline="\n",
    )
    assert solite_cli(["run", "a.sql"], cwd=tmp_path).stdout == snapshot


def test_run_basic_dots(solite_cli, snapshot, tmp_path):
    (tmp_path / "a.sql").write_text(
        """
.timer off

select 1;

.print yo

.print yo2
""",
        newline="\n",
    )
    assert solite_cli(["run", "a.sql"], cwd=tmp_path).stdout == snapshot


def test_run_error(solite_cli, snapshot, tmp_path):
    (tmp_path / "a.sql").write_text(
        """
.timer off

select 'hello' as world;
select 1 + 1 as result;

select
  1,
  2,
  3,
  substr(),
  4,
  5,
  6;

""",
        newline="\n",
    )
    assert solite_cli(["run", "a.sql"], cwd=tmp_path).stdout == snapshot(name="stdout")
    assert solite_cli(["run", "a.sql"], cwd=tmp_path).stderr == snapshot(name="stderr")
