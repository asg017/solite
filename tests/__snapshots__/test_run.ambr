# serializer version: 1
# name: test_run_basic
  '''
  ✓ timer set off
  a.sql:4:1 select 1;
  ┌───┐
  │ 1 │
  ├───┤
  │ 1 │
  └───┘
  a.sql:5:1 select 2;
  ┌───┐
  │ 2 │
  ├───┤
  │ 2 │
  └───┘
  a.sql:7:1 select 3
  ┌───┐
  │ 3 │
  ├───┤
  │ 3 │
  └───┘
  
  '''
# ---
# name: test_run_basic_dots
  '''
  ✓ timer set off
  a.sql:4:1 select 1;
  ┌───┐
  │ 1 │
  ├───┤
  │ 1 │
  └───┘
  yo
  yo2
  
  '''
# ---
# name: test_run_error[stderr]
  '''
  SQL logic error (1)
   8 |   1,
   9 |   2,
  10 |   3,
  11 |   substr(),
     |   ^ wrong number of arguments to function substr()
  12 |   4,
  13 |   5,
  14 |   6;
  	at a.sql:11:3
  
  
  '''
# ---
# name: test_run_error[stdout]
  '''
  ✓ timer set off
  a.sql:4:1 select 'hello' as world;
  ┌───────┐
  │ world │
  ├───────┤
  │ hello │
  └───────┘
  a.sql:5:1 select 1 + 1 as result;
  ┌────────┐
  │ result │
  ├────────┤
  │      2 │
  └────────┘
  
  '''
# ---
